lib/fromZodError.ts

```diff
@@ -22,12 +23,22 @@ export function fromZodError(
   zodError: ZodError,
   options: FromZodErrorOptions = {}
 ): ValidationError {
-  if (!(zodError instanceof zod.ZodError)) {
+  // perform runtime check to ensure the input is a ZodError
+  // why? because people have been historically using this function incorrectly
+  // and then complain on github that it doesn't work
+  if (!isZodErrorLike(zodError)) {
     throw new TypeError(
       `Invalid zodError param; expected instance of ZodError. Did you mean to use the "${fromError.name}" method instead?`
     );
   }
 
+  return fromZodErrorWithoutRuntimeCheck(zodError, options);
+}
+
+export function fromZodErrorWithoutRuntimeCheck(
```

Comment:
dev: sometimes we want to avoid the additional runtime check, e.g. when calling this function from within `toValidationError` we already know that the error instance is of `ZodError`

Reply:
Good call